runtime error: invalid memory address or nil pointer dereference

Stack trace from the terraform-provider-mailtrap_v0.0.2 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbd3fdf]

goroutine 38 [running]:
github.com/yardstick/terraform-provider-mailtrap/api.handleError(0xc0002fb0e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
github.com/yardstick/terraform-provider-mailtrap/api/client.go:115 +0x7f
github.com/yardstick/terraform-provider-mailtrap/api.(*Client).Get(0xc0002c8890, 0xc0001e5998, 0x10, 0xc0000165d2, 0x7, 0xc0001e5998)
github.com/yardstick/terraform-provider-mailtrap/api/client.go:28 +0xce
github.com/yardstick/terraform-provider-mailtrap/provider.resourceReadInbox(0xc0000e37a0, 0xce6d20, 0xc0002c8890, 0xc0000e37a0, 0x0)
github.com/yardstick/terraform-provider-mailtrap/provider/resource_inbox.go:76 +0xc5
github.com/hashicorp/terraform/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000212a80, 0xc0000acaf0, 0xce6d20, 0xc0002c8890, 0xc000206480, 0xf2b5c0, 0xc000206480)
github.com/hashicorp/terraform@v0.14.8/helper/schema/resource.go:466 +0x14b
github.com/hashicorp/terraform/helper/plugin.(*GRPCProviderServer).ReadResource(0xc00020a900, 0xf2a700, 0xc0005cdb60, 0xc00006d080, 0xc00020a900, 0xc0005cdb60, 0xc0003acba0)
github.com/hashicorp/terraform@v0.14.8/helper/plugin/grpc_provider.go:543 +0x407
github.com/hashicorp/terraform/internal/tfplugin5._Provider_ReadResource_Handler(0xd68960, 0xc00020a900, 0xf2a700, 0xc0005cdb60, 0xc00006d020, 0x0, 0xf2a700, 0xc0005cdb60, 0xc000514000, 0xb6)
github.com/hashicorp/terraform@v0.14.8/internal/tfplugin5/tfplugin5.pb.go:3267 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00027a540, 0xf34040, 0xc0002e3500, 0xc0003bd100, 0xc0002d29c0, 0x158cb10, 0x0, 0x0, 0x0)
google.golang.org/grpc@v1.31.1/server.go:1180 +0x50a
google.golang.org/grpc.(*Server).handleStream(0xc00027a540, 0xf34040, 0xc0002e3500, 0xc0003bd100, 0x0)
google.golang.org/grpc@v1.31.1/server.go:1503 +0xcfd
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0002101f0, 0xc00027a540, 0xf34040, 0xc0002e3500, 0xc0003bd100)
google.golang.org/grpc@v1.31.1/server.go:843 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.31.1/server.go:841 +0x204

Error: The terraform-provider-mailtrap_v0.0.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/api/client.go
+++ b/api/client.go
@@ -112,11 +112,11 @@ func handleError(resp *http.Response, json []byte, err error) error {
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		defer resp.Request.Body.Close()
-		reqBody, _ := ioutil.ReadAll(resp.Request.Body)
+		defer resp.Body.Close()
+		respBody, _ := ioutil.ReadAll(resp.Body)
 		log.Error("NON-OK Status Code for Request")
 		log.Error("URL: " + resp.Request.URL.String())
-		log.Error("BODY: " + string(reqBody))
+		log.Error("BODY: " + string(respBody))
 		log.Error("STATUS: " + resp.Status)
 		return errors.New("NON-OK Status Code Returned From Mailtrap")
 	}

